### PR TITLE
Account check printing report base by journal

### PR DESCRIPTION
**Pull request by andres-pcg**
This improvement is to allow checks layouts to be asigned by journal instead of company for multi banks companies.

### CODE REVIEW

**Comment by andres-pcg** [comment]
This might take care of #200

File: account_check_printing_report_base/models/res_config_settings.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+# Copyright 2018 Iterativo - Manuel Marquez <buzondemam@gmail.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import api, fields, models
+
+
+class ResConfigSettings(models.TransientModel):
+    _inherit = 'res.config.settings'
+
+    check_layout_verification = fields.Selection([
```
Comment (pedrobaeza) [inline account_check_printing_report_base/models/res_config_settings.py]: Don't add this option, as it's simpler: if you put a layout in the journal, then use that one. If not, use the company's one.

File: account_check_printing_report_base/models/account_payment.py [hunk 2]
```diff
@@ -27,11 +27,36 @@ class AccountPayment(models.Model):
...
+        check_layout_verification = ICPSudo.get_param(
+            'account_check_printing_report_base.check_layout_verification')
+
         for rec in self:
-            if rec.company_id.check_layout_id:
-                return self.env['ir.actions.report']._get_report_from_name(
-                    rec.company_id.check_layout_id.report
-                ).report_action(self)
+
+            if not check_layout_verification or \
```
Comment (etobella) [inline account_check_printing_report_base/models/account_payment.py]: as @pedrobaeza said, you can remove this if and maintain only the last option. You should modify the check with this logic.

**Review by etobella — CHANGES_REQUESTED** [review]
It is a great idea, but some modifications should be addressed!

**Comment by mamcode** [comment]
@pedrobaeza @etobella Please review again.

**Comment by pedrobaeza** [comment]
There are conflicts in the PR.

**Comment by mamcode** [comment]
@pedrobaeza Conflicts resolved. Please review.

File: account_check_printing_report_base/__manifest__.py [hunk 3]
```diff
@@ -2,15 +2,17 @@
...
 
 {
     "name": "Account Check Printing Report Base",
-    "version": "11.0.1.1.1",
+    "version": "11.0.1.2.0",
     "license": "AGPL-3",
     "author": "Eficent,"
               "Serpent Consulting Services Pvt. Ltd.,"
               "Ursa Information Systems,"
+              "iterativo SRL,"
```
Comment (pedrobaeza) [inline account_check_printing_report_base/__manifest__.py]: @etobella @jbeficent judge if you consider this new feature enough for granting co-authorship

Comment (gustavovalverde) [reply account_check_printing_report_base/__manifest__.py]: @pedrobaeza is more urgent for us to have the feature available than the authorship. If that's a problem, please just help us with the merge.

Comment (pedrobaeza) [reply account_check_printing_report_base/__manifest__.py]: OK, then please remove it meanwhile. Don't forget any way to add you at least as contributor in the README. You deserve this.

Comment (gustavovalverde) [reply account_check_printing_report_base/__manifest__.py]: @mamcode please proceed as stated by Pedro

Comment (mamcode) [reply account_check_printing_report_base/__manifest__.py]: Done!

**Pull request merged**